src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml

```diff
@@ -64,9 +64,52 @@
   tags:
     - new_ansible_sudo_pass
 
+- name: lookup ssh pubkeys from authorized_keys_dir and create ssh_pubkeys_list
+  set_fact:
+    ssh_pubkeys_list: "{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}"
+  with_items:
+    "{{ dev_users.present }}"
+  register: ssh_pubkeys_results_list
+  tags:
+    - pubkey
+
+- name: iterate over ssh_pubkeys_list and join into a string
+  set_fact:
+    ssh_pubkeys_string: "{{ ssh_pubkeys_results_list.results | map(attribute='ansible_facts.ssh_pubkeys_list') | list | join('\n') }}"
+  tags:
+    - pubkey
+
+- name: capturing username
+  pause:
+    prompt: "Please enter your Dimagi username"
+  register: username_input
```

Comment:
Who is "your" referring to in this prompt? I'm confused about how this helps with removing retired keys from the authorized keys file since (a) there may be multiple retired keys and (b) the person running ansible will rarely, if ever, be the person who's key is being retired.

Will this prompt be presented every time any command that includes this role (other than `deploy_stack`) is run (ex: during `cchq ... update-users` or `cchq ... update-user-key`)? Will it be necessary to enter a username even when no keys have been retired?

Reply:
@millerdev "your" refers to the person who is running the `update-users` or `update-user-key` command. We would like to ensure that the "entered" user's key is not removed from the list of ssh-keys while updating the authorized_keys file. That user should be valid dev_users.present. 
yes, It will prompt for both update-users and update-user-key commands. 

@dannyroberts Tested in the following ways
1. deploy-stack --first-time
2. deploy-stack --skip-check
3. update-users - with valid dev_user name
4. update-users - without valid dev_user name
5. update-user-key ( found a bug today while testing again for screenshots, explanation below).
    - with/without valid dev_user name. 
    
Bug: Problem with `exclusive: yes` for `[bootstrap-users: Authorize current devs to login as ansible]` task. 
since update-users and update-user-key both runs `tags: pubkey` tasks. the `exclusive: yes` will overwrite the ansible authorized_keys file and adds only one key into it when the `update-user-key <username>` command is executed.
Sorry for missing this. I was more concentrating on the user_prompt testing. 
@millerdev thank you. Now, I understood your comment (ticket) on why you mentioned that we may have to rewrite or modify the logic for both commands. 

@dannyroberts I have a go through the tasks again. So, I will get back to you on this. 
Thanks.

Reply:
@millerdev yeah to put it another way, one of the risks we identified is that someone might somehow inadvertently remove their own key. To mitigate that risk, we use this task to ask the user running the command to input their own ssh username and then check to make sure that that username shows up in the exclusive list of keys we will be overwriting the authorized_keys file with. That makes it really hard for someone to inadvertently remove their own key.

We believe everything would work fine without this check as long as no one made any mistakes, but having this check helps prevent such a mistake.